### PR TITLE
categorical choropleths

### PR DESCRIPTION
**Pull request by nicolaskruchten**

### CODE REVIEW

**Comment by nicolaskruchten** [comment]
missing docs, of course

File: packages/python/plotly/plotly/express/_chart_types.py [hunk 1]
```diff
@@ -866,7 +870,13 @@ def choropleth(
     return make_figure(
         args=locals(),
         constructor=go.Choropleth,
-        trace_patch=dict(locationmode=locationmode),
+        trace_patch=dict(
+            locationmode=locationmode,
+            featureidkey=featureidkey,
+            geojson=geojson
+            if not hasattr(geojson, "__geo_interface__")
```
Comment (emmanuelle) [inline packages/python/plotly/plotly/express/_chart_types.py]: add a comment here to explain why this is needed?

Comment (nicolaskruchten) [reply packages/python/plotly/plotly/express/_chart_types.py]: sure, this is a special hook for `geopandas`... if you have a geopandas frame, it has a special property `.geometry` which has a special `.__geo_interface__` property which returns GeoJSON, so doing this here allows the user to say `px.choropleth(df, geojson=df.geometry,...)`

Comment (emmanuelle) [reply packages/python/plotly/plotly/express/_chart_types.py]: thanks, I was wondering whether this explanation could be added to the source code as a comment, for future maintainers

**Comment by emmanuelle** [comment]
Looks good to me, and the example with the `gapminder` and `continent` color will be straightforward to add. It would be good practice to add it here but it can also be another PR, so :dancer: for this part. The only issue I experienced is that when you hide a legend item you get the "double-click on the legend to isolate a trace" message, which I do not really understand (or it does not work), but this is not related to this PR.

**Comment by nicolaskruchten** [comment]
> the "double-click on the legend to isolate a trace" message, which I do not really understand (or it does not work)

it does work for me... although it is odd that it's just that one hint we give to users

**Comment by emmanuelle** [comment]
:dancer:

**Comment by emmanuelle** [comment]
:dancer:

**Pull request merged**